src/status_im/network/core.cljs

```diff
@@ -97,24 +106,62 @@
   {:db       (assoc db :networks/manage (validate-manage default-manage))
    :dispatch [:navigate-to :edit-network]})
 
-(fx/defn connect [{:keys [db now] :as cofx} {:keys [network on-success on-failure]}]
-  (if (get-in db [:account/account :networks network])
-    (let [current-network (get-in db [:account/account :networks (:network db)])]
-      (if (ethereum/network-with-upstream-rpc? current-network)
-        (fx/merge cofx
-                  #(action-handler on-success network %)
-                  (accounts.update/account-update
-                   {:network      network
-                    :last-updated now}
-                   {:success-event [:accounts.update.callback/save-settings-success]}))
-        (fx/merge cofx
-                  {:ui/show-confirmation {:title               (i18n/label :t/close-app-title)
-                                          :content             (i18n/label :t/close-app-content)
-                                          :confirm-button-text (i18n/label :t/close-app-button)
-                                          :on-accept           #(re-frame/dispatch [:network.ui/save-non-rpc-network-pressed network])
-                                          :on-cancel           nil}}
-                  #(action-handler on-success network %))))
-    (action-handler on-failure)))
+(fx/defn connect-success [{:keys [db now] :as cofx} {:keys [network-id on-success client-version]}]
+  (let [current-network (get-in db [:account/account :networks (:network db)])]
+    (if (ethereum/network-with-upstream-rpc? current-network)
+      (fx/merge cofx
+                #(action-handler on-success {:network-id network-id :client-version client-version} %)
+                (accounts.update/account-update
+                 {:network      network-id
+                  :last-updated now}
+                 {}))
+      (fx/merge cofx
+                {:ui/show-confirmation {:title               (i18n/label :t/close-app-title)
+                                        :content             (i18n/label :t/close-app-content)
+                                        :confirm-button-text (i18n/label :t/close-app-button)
+                                        :on-accept           #(re-frame/dispatch [:network.ui/save-non-rpc-network-pressed network-id])
+                                        :on-cancel           nil}}
+                #(action-handler on-success {:network-id network-id :client-version client-version} %)))))
+
+(defn connect-failure [{:keys [network-id on-failure reason]}]
+  (action-handler on-failure
+                  {:network-id network-id :reason reason}
+                  nil))
+
+(fx/defn connect [{:keys [db]} {:keys [network-id on-success on-failure]}]
+  (if-let [config (get-in db [:account/account :networks network-id :config])]
+    (do
+      (http/post (get-in config [:UpstreamConfig :URL])
```

Comment:
Sorry but this is big no-no, why are we doing side effect action in handler function which is supposed to be pure ?
This function can't be unit tested for example, because it will crash!
We already have `:http/post` effect, so I'm puzzled why we did this.

Reply:
`:http-post` actually

Reply:
@janherich does it look better now?

Reply:
That's awesome, thank you very much @rasom !